data/txHandler.go

```diff
@@ -328,19 +414,134 @@ type SolicitedTxHandler interface {
 	Handle(txgroup []transactions.SignedTxn) error
 }
 
+type solicitedTxHandler struct {
+	txHandler *TxHandler
+}
+
 // SolicitedTxHandler converts a transaction handler to a SolicitedTxHandler
 func (handler *TxHandler) SolicitedTxHandler() SolicitedTxHandler {
 	return &solicitedTxHandler{txHandler: handler}
 }
 
-type solicitedTxHandler struct {
-	txHandler *TxHandler
-}
-
 func (handler *solicitedTxHandler) Handle(txgroup []transactions.SignedTxn) error {
-	outmsg, _ := handler.txHandler.processDecoded(txgroup)
-	if outmsg.Action == network.Disconnect {
+	disconnect := handler.txHandler.processDecoded(txgroup)
+	if disconnect {
 		return fmt.Errorf("invalid transaction")
 	}
 	return nil
 }
+
+// SolicitedAsyncTxHandler handles messages received through channels other than the gossip network.
```

Comment:
Why can't txnsync use the synchronous txn handler?

Reply:
the txsync is implemented as a single threaded component. We don't want to block the processing of incoming messages while the verification process takes place.